ballerina/model-provider.bal

```diff
@@ -216,37 +216,34 @@ public isolated distinct client class Wso2ModelProvider {
                     assistantMessage["content"] = message?.content;
                 }
                 chatCompletionRequestMessages.push(assistantMessage);
-            } else if message is ChatUserMessage|ChatSystemMessage {
-                intelligence:ChatCompletionRequestMessage trasnformedMessage = self.mapUserOrSystemMessage(message);
+                continue;
+            }
+            if message is ChatUserMessage|ChatSystemMessage {
+                intelligence:ChatCompletionRequestMessage transformedMessage = self.mapUserOrSystemMessage(message);
                 if message.name is string {
-                    trasnformedMessage["name"] = message.name;
+                    transformedMessage["name"] = message.name;
                 }
-                chatCompletionRequestMessages.push(trasnformedMessage);
-            } else {
-                chatCompletionRequestMessages.push(message);
+                chatCompletionRequestMessages.push(transformedMessage);
+                continue;
             }
+            chatCompletionRequestMessages.push(message);
         }
         return chatCompletionRequestMessages;
     }
 
     private isolated function mapToFunctionCall(intelligence:ChatCompletionFunctionCall functionCall)
     returns FunctionCall|LlmError {
         do {
-            json jsonArgs = check functionCall.arguments.fromJsonString();
-            map<json>? arguments = check jsonArgs.cloneWithType();
+            map<json>? arguments = check functionCall.arguments.fromJsonWithType();
```

Comment:
This isn't the same btw. `fromJsonWithType` should fail when the value is a string. 

```ballerina
import ballerina/io;

type Input record {|
    int id;
    string[] interests;
|};

public function main() {
    json j = {id: 1, interests: ["reading", "traveling"]};
    string js = j.toJsonString();

    io:println(js.fromJsonWithType(Input));
    io:println(js.fromJsonStringWithType(Input));
}
```

```
error("{ballerina/lang.value}ConversionError",message="'string' value cannot be converted to 'Input'")
{"id":1,"interests":["reading","traveling"]}
```

Given that `arguments` is a string (https://github.com/ballerina-platform/module-ballerina-ai/blob/main/ballerina/modules/intelligence/types.bal#L207), I'm not sure how this works.